usdigitalresponse/cpf-reporter
# Feature/2228 Add Agencies page

Not quite ready for the review, but feedback welcome!

### Quick summary
I added Agencies page (and other CRUD pages generated by Redwood, some commented out as they're part of other tickets). Agencies are displayed if they're associated with, temporarily hardcoded, tenant with id 2. 
This page should also be locked for admin users only, but I wasn't able to mock users.

<img width="1120" alt="Screenshot 2023-12-01 at 12 36 18 PM" src="https://github.com/usdigitalresponse/cpf-reporter/assets/20152824/f5ef5654-31c6-45ab-9dd9-ee4abcbf481c">

### Features

Agencies Page
- [ ] Can be viewed only by admin users
- [x] Has title
- [x] Has Create New Agency button
- [x] Has Table
    - [x] Displays list of agencies
        - [x] Only agencies associated with the tenant of the user - currently hardcoded tenant id
     - [x] Agency Cell
        - [x] Shows Agency Code
        - [x] Shows Agency Name
        - [x] Shows Edit button

### Prisma
I added very basic models for `Agency` and `Tenant` to `schema.prisma`. It might be a better idea for one person to generate all models, as redwood has a scaffolding tool that allows to create all CRUD operations and routes.

### Styling
Redwood seems to like Tailwind, but TailwindUI that offers premade components is a paid service. I added Bootstrap with all overrides that could be found in GOST.
The design for the page attached to the ticket didn't seem to follow the design system from Figma: I replaced the Bootstrap colors for buttons with colors USDR uses. Wasn't sure about the table design, I followed the design of "Grants" table.

### Users
I attempted to create users, but it seems like the whole authentication should be finished before using `<Private>` routes and I couldn't mock or hardcode a user temporarily.

Quick note: I ran into `Command 'npm' not found` when trying to run `yarn rw setup auth dbAuth`, it can be resolved by upgrading yarn according to these instructions (https://yarnpkg.com/migration/guide)

### Testing the functionality
1. Run `yarn rw prisma studio` to open a web app view of the database
2. You'll be able to create new tenants and agencies. Assign an agency to a tenant with id 2.
3. Verify that http://localhost:8910/agencies only shows agencies that belong to that tenant. Note: using `Create new agency` button will not work as the new agency will not be assigned to any tenants (no users yet).

<img width="1466" alt="Screenshot 2023-12-01 at 12 47 05 PM" src="https://github.com/usdigitalresponse/cpf-reporter/assets/20152824/75e8caa1-5b00-4594-bbc9-470c426e0feb">
